tachyon/zk/plonk/permutation/permutation_keys.h

```diff
@@ -0,0 +1,117 @@
+#ifndef TACHYON_ZK_PLONK_PERMUTATION_KEYS_H_
+#define TACHYON_ZK_PLONK_PERMUTATION_KEYS_H_
+
+#include "tachyon/base/buffer/copyable.h"
+#include "tachyon/export.h"
+#include "tachyon/math/elliptic_curves/affine_point.h"
+#include "tachyon/math/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon {
+namespace zk {
+
+template <typename Curve>
+class PermutationVerifyingKey;
+
+template <typename Curve>
+class PermutationVerifyingKey {
+ public:
+  using CommitmentTy = math::AffinePoint<Curve>;
+  using CommitmentVec = std::vector<CommitmentTy>;
+
+  PermutationVerifyingKey() = default;
+
+  PermutationVerifyingKey(const CommitmentVec& commitments)
+      : commitments_(commitments) {}
+
+  PermutationVerifyingKey(CommitmentVec&& commitments)
+      : commitments_(std::move(commitments)) {}
+
+  const CommitmentVec& commitments() const { return commitments_; }
+
+  size_t BytesLength() const { return base::EstimateSize(commitments_); }
+
+ private:
+  CommitmentVec commitments_;
+};
+
+template <typename F, size_t MaxDegree>
+class PermutationProvingKey {
+ public:
+  using DensePoly = math::UnivariateDensePolynomial<F, MaxDegree>;
+  using Evals = math::UnivariateEvaluations<F, MaxDegree>;
+
+  PermutationProvingKey() = default;
+
+  PermutationProvingKey(const std::vector<Evals>& permutations,
+                        const std::vector<DensePoly>& polys)
+      : permutations_(permutations), polys_(polys) {}
+
+  PermutationProvingKey(std::vector<Evals>&& permutations,
+                        std::vector<DensePoly>&& polys)
+      : permutations_(std::move(permutations)), polys_(std::move(polys)) {}
+
+  size_t BytesLength() const { NOTIMPLEMENTED(); };
```

Comment:
Why is that?

Reply:
it is missing implementation.. i will resolve this!